Portal/src/Datahub.Portal/Pages/Workspace/Reports/WorkspaceCostsTable.razor

```diff
@@ -0,0 +1,103 @@
+﻿@inject IDbContextFactory<DatahubProjectDBContext> _dbContextFactory
+@inject IJSRuntime _jsRuntime
+
+<MudTable Items="Costs" Dense="true" Outlined="true" RowsPerPage="10" Striped="true">
+    <ToolBarContent>
+        <MudText Typo="Typo.h2">
+            @Localizer["Your workspace's daily costs"]
+        </MudText>
+        <MudSpacer/>
+        <DHButton Variant="Variant.Filled" Color="Color.Primary" StartIcon="@SidebarIcons.Download" OnClick="HandleDownload" Class="mr-4">
+            @Localizer["Download to CSV"]
+        </DHButton>
+    </ToolBarContent>
+    <HeaderContent>
+        <MudTh>
+            <strong>@Localizer["Date"]</strong>
+        </MudTh>
+        <MudTd>
+            <strong>@Localizer["Source"]</strong>
+        </MudTd>
+        <MudTd>
+            <strong>@Localizer["Cost"]</strong>
+        </MudTd>
+    </HeaderContent>
+    <RowTemplate>
+        <MudTd DataLabel="@Localizer["Date"]">
+            @context.Date
+        </MudTd>
+        <MudTd DataLabel="@Localizer["Source"]">
+            @context.Source
+        </MudTd>
+        <MudTd DataLabel="@Localizer["Cost"]">
+            @(context.Cost < (decimal)0.01 ? $"< {(0.01).ToString("C")}" : context.Cost.ToString("C"))
+        </MudTd>
+    </RowTemplate>
+    <PagerContent>
+        <MudTablePager PageSizeOptions="new[] { 10, 50, 100, int.MaxValue }"/>
+    </PagerContent>
+</MudTable>
+<script>
+    window.downloadFileFromStream = async (fileName, contentStreamReference) => {
+        const arrayBuffer = await contentStreamReference.arrayBuffer();
+        const blob = new Blob([arrayBuffer]);
+        const url = URL.createObjectURL(blob);
+        const anchorElement = document.createElement('a');
+        anchorElement.href = url;
+        anchorElement.download = fileName ?? '';
+        anchorElement.click();
+        anchorElement.remove();
+        URL.revokeObjectURL(url);
+    }
+</script>
```

Comment:
Can we move this into a co-located js file please?

I'd like to leverage the ES6 modules instead of attaching to the window object. There should be some solid references in the storage explorer files (don't forget to dispose!)